lib/src/patterns.dart

```diff
@@ -40,10 +40,24 @@ final hrPattern = RegExp(r'^ {0,3}([-*_])[ \t]*\1[ \t]*\1(?:\1|[ \t])*$');
 /// the list marker.
 final ulPattern = RegExp(r'^([ ]{0,3})()([*+-])(([ \t])([ \t]*)(.*))?$');
 
+/// Similar to [ulPattern] but with a GitHub style checkbox
+/// '[ ]'|'[x]'|'[X]' following the number. The checkbox will
+/// be grabbed by group [5] and ulPattern's groups [5,6,7] are all
+/// shifted 2 places to be [7,8,9]
+final ulWithCheckBoxPattern = RegExp(
+    r'^([ ]{0,3})()([*+-])([ \t]{0,4})(\[[ xX]{1}\])(([ \t])([ \t]*)(.*))?$');
+
 /// A line starting with a number like `123.`. May have up to three leading
 /// spaces before the marker and any number of spaces or tabs after.
 final olPattern = RegExp(r'^([ ]{0,3})(\d{1,9})([\.)])(([ \t])([ \t]*)(.*))?$');
 
+/// Similar to [olPattern] but with a GitHub style checkbox
+/// '[ ]'|'[x]'|'[X]' following the number. The checkbox will
```

Comment:
This regular expression should probably be in backticks, so as to not look like a markdown reference.